From 429db6536754dcfec421539cce6c33f9b990f1dc Mon Sep 17 00:00:00 2001
From: Chao Liu <chao.liu@processmission.com>
Date: Fri, 9 Oct 2026 22:32:17 +0800
Subject: [PATCH] arch: arm64: place interrupt metadata at the virtual range
 end

Allow RAM regions to cross 4 GiB when a wider virtual address range is
selected. Keep the interrupt metadata outside the live RAM region in
the intermediate ELF used to generate ISR tables.

Signed-off-by: Chao Liu <chao.liu@processmission.com>
---
 include/zephyr/arch/arm64/scripts/linker.ld | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/zephyr/arch/arm64/scripts/linker.ld b/include/zephyr/arch/arm64/scripts/linker.ld
index d6799e60e..6775e4523 100644
--- a/include/zephyr/arch/arm64/scripts/linker.ld
+++ b/include/zephyr/arch/arm64/scripts/linker.ld
@@ -78,7 +78,7 @@ MEMORY
     RAM       (wx) : ORIGIN = RAM_ADDR, LENGTH = RAM_SIZE
     LINKER_DT_REGIONS()
     /* Used by and documented in include/linker/intlist.ld */
-    IDT_LIST  (wx) : ORIGIN = 0xFFFF8000, LENGTH = 32K
+    IDT_LIST  (wx) : ORIGIN = (1 << CONFIG_ARM64_VA_BITS) - 32K, LENGTH = 32K
 #ifdef CONFIG_LINKER_USE_ONDEMAND_SECTION
     ONDEMAND (rx) : ORIGIN = (CONFIG_KERNEL_VM_BASE), LENGTH = (CONFIG_KERNEL_VM_SIZE)
 #endif
-- 
2.56.0

